### ISSUE TITLE
Infix notation for functions

### ISSUE THREAD

**Issue by McEarl** [Naproche/Naproche-SAD#14]
Naproche doesn't allow infix notation for functions.

Examples:

- `Definition. Let a,b be objects and f be a function. Assume that (a,b) is an element of Dom(f). a f b = f[(a,b)].`
- `Let a f b stand for f[(a,b)].`

Both the definition and the let-statement lead to an error which states `unexpected f`.

This error can be avoided by writing something like `a 'f' b` instead of `a f b`, but this is not a very natural writing style for mathematical texts.

**Comment by anfelor** [comment]
>  error can be avoided by writing something like a 'f' b instead of a f b,

Exactly! I think a better way to describe this issue is that Naproche doesn't allow for *alpha-numeric* infix functions. You can still use `+` or `>>` as infixes.

**Comment by McEarl** [comment]
I've tried to use `+` as an infix function but the only way it worked was declaring `+` as a signature extension:

`Signature. Let a,b be objects. a + b is an object.`

In contrast the statement `Let a + b stand for +[(a,b)].` generates the error `unexpected .` and defining `+` as a function, i.e.

`Definition. Let a,b be objects and + be a function. Assume that (a,b) is an element of Dom(+). a + b = +[(a,b)].`,

generates the error

`unexpected +`
`expecting such-that attribute, that clause, is predicate, isA predicate, does predicate, atomic statement, simple statement, symbolic term or a dot`.



But Naproche doesn't seem to accept non-alphanumeric characters as variables in any case (except in signature extensions). For example

`Lemma. Let + be a function. Dom(+) is a set.`

generates the error

`unexpected +`
`expecting chained statement`.
